### PR TITLE
#169649584 bookmark and get bookmarked accommodations

### PR DESCRIPTION
**Pull request by lemoissonM**
### What does this PR do?
- bookmark accommodations
- undo accommodation
- get bookmarked accommodations
#### Description of Task to be completed?
- create tests
- create bookmark service 
- create routes and controllers functions 
- create a bookmark table
#### How should this be manually tested?
- npm run dev
- POST api/v1/accommodations/:AccommodationId/bookmark to bookmark or undo bookmark
- GET api/v1/accommodations/bookmarked to get bookmarked accommodations
- These two endpoints require a token
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/169649584
#### Screenshots (if appropriate)
Screen Shot 2019-11-11 at 12 43 46
Screen Shot 2019-11-11 at 12 43 55
Screen Shot 2019-11-11 at 12 44 13

#### Questions:

### CODE REVIEW

File: src/routes/accommodationRoute.js [hunk 1]
```diff
@@ -83,21 +83,23 @@ const fUpload = fileUpload({
...
 app.get('/:accommodationId', AccommodationsController.getAccommodationById);
 app.post('/:accommodationId/like', [
   checkToken,
   LikingsController.addLikeAccommdation
 ]);
-
-
+app.post('/:accommodationId/bookmark', [checkToken, valid.accommodationId,
+  accommodationExists,
+  AccommodationsController.bookmarkAccommodation]);
```
Comment (urbanishimwe) [inline src/routes/accommodationRoute.js]: nice job so far.
However, as the story says `enable bookmarking` which also includes `un-bookmarking` I suggest you separate the functionalities for bookmarking and un-bookmarking into different routes.
it won't take many functionalities.

File: src/services/bookmark.service.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  async bookmark(data) {
+    const isBookmarked = await db.Bookmarks.findOne({ where: data });
+    let result;
+    if (!isBookmarked) {
+      await db.Bookmarks.create(data);
+      result = 'added';
+    } else {
+      await db.Bookmarks.destroy({ where: data });
+      result = 'deleted';
```
Comment (urbanishimwe) [inline src/services/bookmark.service.js]: `post` is for creating and `delete` is for deleting!
Imagine a user made a `post` request and receive a message saying `Bookmark deleted successfully`! it won't make sense!
please try to separate these functionalities bruh!

File: src/controllers/accommodationController.js [hunk 3]
```diff
@@ -106,5 +107,20 @@ const Accommodation = {
...
   },
 
+  async bookmarkAccommodation(req, res) {
+    const { userId } = req.userData;
+    const { accommodationId } = req.params;
+
+    const bookmarkData = { UserId: userId, AccommodationId: accommodationId };
+    const result = await BookmarkService.bookmark(bookmarkData);
+    const status = result === 'added' ? 201 : 200;
+    return sendResult(res, status, `Bookmark ${result} successfuly`);
```
Comment (urbanishimwe) [inline src/controllers/accommodationController.js]: `Successfully` not `successfuly` this is a typo!

**Review by urbanishimwe — CHANGES_REQUESTED** [review]

File: src/controllers/accommodationController.js (hunk 3 above)
Comment (lemoissonM) [reply src/controllers/accommodationController.js]: okey

File: src/services/bookmark.service.js (hunk 2 above)
Comment (lemoissonM) [reply src/services/bookmark.service.js]: okey

File: src/routes/accommodationRoute.js (hunk 1 above)
Comment (lemoissonM) [reply src/routes/accommodationRoute.js]: great!

**Review by swaibat — APPROVED** [review]

**Review by urbanishimwe — APPROVED** [review]

File: src/routes/accommodationRoute.js [hunk 4]
```diff
@@ -17,21 +17,26 @@ const fUpload = fileUpload({
...
+  isSupplierAccommodation, checkForImages, checkForImagesUpdate, accommodationExists
 } = AccommodationMiddleware;
 const { checkToken } = UserMiddleware;
 
 app.patch('/:id', fUpload, checkToken, RoleMiddleware.checkHostOrTAdmin, isSupplierAccommodation, checkForImagesUpdate, valid.editAccommodation, AccommodationsController.editAccommodation);
 app.delete('/:id', checkToken, RoleMiddleware.checkHost, isSupplierAccommodation, AccommodationsController.deleteAccommodation);
 app.post('/', fUpload, checkToken, RoleMiddleware.checkHostOrTAdmin, valid.accommodation, checkForImages, AccommodationsController.addAccommodation);
 app.get('/', AccommodationsController.getAccommodations);
-app.get('/:accommodationId/rating', valid.getReviewvalidation, BookingsController.getRating);
+app.get('/:accommodationId/rating', valid.accommodationId, BookingsController.getRating);
```
Comment (elchroy) [inline src/routes/accommodationRoute.js]: Why this change?

File: src/routes/accommodationRoute.js [hunk 5]
```diff
@@ -17,21 +17,26 @@ const fUpload = fileUpload({
...
   LikingsController.addLikeAccommdation
 ]);
-
-
+app.post('/:accommodationId/bookmark', [checkToken, valid.accommodationId,
+  accommodationExists,
+  AccommodationsController.bookmarkAccommodation]);
+app.delete('/:accommodationId/bookmark', [checkToken, valid.accommodationId,
+  accommodationExists,
+  AccommodationsController.undoBookmark]);
```
Comment (elchroy) [inline src/routes/accommodationRoute.js]: We have a format that we've been using, yeah? Why this new array format for middlewares?
